### PR TITLE
Migrate to null safety + PC 3.0.0

### PR DESCRIPTION
**Pull request by acoutts**
Builds on https://github.com/appsup-dart/crypto_keys/pull/3 with additional migration to null safety.

There is just one test failing:
```
00:02 +8 -1: test/crypto_keys_test.dart: Signing Signing with EC keys Example Signing Using ECDSA P-256K SHA-256 [E]                                                                                       
  type 'Null' is not a subtype of type 'List<int>' of 'seed'
  dart:core                                                          _TypeError._throwNew
  package:pointycastle/src/ec_standard_curve_constructor.dart 20:21  constructFpStandardCurve
  package:pointycastle/ecc/curves/secp256k1.dart 15:34               new ECCurve_secp256k1
  package:crypto_keys/src/asymmetric_operator.dart 10:19             _AsymmetricOperator.createCurveParameters
  package:crypto_keys/src/keys.dart 86:33                            new KeyPair.generateEc
  test/crypto_keys_test.dart 903:31                                  main.<fn>.<fn>.<fn>
  
00:03 +25 -1: Some tests failed.  
```

One are I wasn't sure on is when using `ParametersWithRandom` from PC you cannot pass `null` for `random` anymore. I took a guess but this might be wrong.

### CODE REVIEW

File: lib/src/asymmetric_operator.dart [hunk 1]
```diff
@@ -99,12 +99,13 @@ class _AsymmetricVerifier extends Verifier<PublicKey>
       : super._(algorithm, key);
 
   @override
-  pc.Signer get _algorithm => super._algorithm;
+  pc.Signer get _algorithm => super._algorithm as pc.Signer;
 
   @override
   bool verify(Uint8List data, Signature signature) {
     if (key is RsaKey) {
-      _algorithm.init(false, pc.ParametersWithRandom(keyParameter, null));
```
Comment (acoutts) [inline lib/src/asymmetric_operator.dart:107]: @jpeiffer @rbellens curious if you know the right way to replace this null now that we can't pass null to PC.

File: lib/src/asymmetric_operator.dart [hunk 2]
```diff
@@ -132,26 +133,25 @@ class _AsymmetricEncrypter extends Encrypter<Key> with _AsymmetricOperator {
...
   @override
-  pc.AsymmetricBlockCipher get _algorithm => super._algorithm;
+  pc.AsymmetricBlockCipher get _algorithm =>
+      super._algorithm as pc.AsymmetricBlockCipher;
 
   @override
   Uint8List decrypt(EncryptionResult input) {
-    _algorithm.init(
-        false,
-        pc.ParametersWithRandom(keyParameter, null //pc.SecureRandom('Fortuna')
```
Comment (acoutts) [inline lib/src/asymmetric_operator.dart:141]: @jpeiffer @rbellens curious if you know the right way to replace this null now that we can't pass null to PC.

**Review by acoutts — COMMENTED** [review]
Need help on these 2 bits.

**Comment by rbellens** [comment]
@acoutts 

Thanks a lot for this pull request.
For the failing test, this was an issue in the pointycastle library. I've created a pull request to fix (see https://github.com/bcgit/pc-dart/pull/81)

**Comment by acoutts** [comment]
That’s great!

**Comment by acoutts** [comment]
Don’t forget to publish on pub as well!

**Pull request merged**